### PR TITLE
feat: add QueryOptions to Spanner samples

### PR DESCRIPTION
**Pull request by olavloite**

### CODE REVIEW

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 1]
```diff
@@ -1433,6 +1434,45 @@ static void queryWithTimestampParameter(DatabaseClient dbClient) {
   }
   // [END spanner_query_with_timestamp_parameter]
 
+  // [START spanner_create_client_with_query_options]
+  static void clientWithQueryOptions(DatabaseId db) {
+    SpannerOptions options =
+        SpannerOptions.newBuilder()
+            .setDefaultQueryOptions(
+                db, QueryOptions.newBuilder().setOptimizerVersion("latest").build())
```
Comment (skuruppu) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: I think here we want to set it to a version that's not the default so changing it to "1" would be ideal.
```suggestion
                db, QueryOptions.newBuilder().setOptimizerVersion("1").build())
```

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java [hunk 2]
```diff
@@ -1433,6 +1434,45 @@ static void queryWithTimestampParameter(DatabaseClient dbClient) {
...
+
+  // [START spanner_query_with_query_options]
+  static void queryWithQueryOptions(DatabaseClient dbClient) {
+    try (ResultSet resultSet =
+        dbClient
+            .singleUse()
+            .executeQuery(
+                Statement
+                    .newBuilder("SELECT SingerId, AlbumId, AlbumTitle FROM Albums")
+                    .withQueryOptions(QueryOptions.newBuilder().setOptimizerVersion("latest").build())
```
Comment (skuruppu) [inline spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Same here, I think it's useful to show how to set the version to something other than the default.

```suggestion
                    .withQueryOptions(QueryOptions.newBuilder().setOptimizerVersion("1").build())
```

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 1 above)
Comment (olavloite) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Done.

File: spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java (hunk 2 above)
Comment (olavloite) [reply spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java]: Done.

**Comment by olavloite** [comment]
@skuruppu This depends on the release of OpenCensus 0.26 where https://github.com/census-instrumentation/opencensus-java/pull/2017 should be included. That would fix https://github.com/googleapis/java-spanner/issues/106.

**Comment by skuruppu** [comment]
CC @mayurkale22. This is one of the critical PRs blocked from being merged at the moment.

**Comment by mayurkale22** [comment]
> CC @mayurkale22. This is one of the critical PRs blocked from being merged at the moment.

I started the release process (https://github.com/census-instrumentation/opencensus-java/pull/2027), will keep you posted once new version is published to maven.

**Comment by olavloite** [comment]
This is now failing on this check: https://github.com/census-instrumentation/opencensus-java/blob/8cc57803bbc3481243cebb8e46e0f389b4b2fdc4/impl_core/src/main/java/io/opencensus/implcore/metrics/DerivedLongCumulativeImpl.java#L96

**Comment by skuruppu** [comment]
@lesv, @gguuss or @chingor13, could we kindly get an approval from one of you for this PR as we're not able to merge it in otherwise.

**Comment by skuruppu** [comment]
> @lesv, @gguuss or @chingor13, could we kindly get an approval from one of you for this PR as we're not able to merge it in otherwise.

Thanks so much @lesv for the approval.

**Pull request merged**